onnxruntime_extensions/tools/pre_post_processing/steps/nlp.py

```diff
@@ -0,0 +1,508 @@
+            for i in range(0, len(self.input_names)):
+                input_type_str_x, input_shape_str_x = self._get_input_type_and_shape_strs(graph, i)
+                inputs.append(f"{input_type_str_x}[{input_shape_str_x}] {self.input_names[i]}")
+            return ",".join(inputs)
+
+        output_shape_str = "any_len"
+        converter_graph = onnx.parser.parse_graph(
+            f"""\
+            tokenizer_decoder ({build_input_declare()}) 
+                => (string[{output_shape_str}] {self.output_names[0]})
+            {{
+                i64_em = Constant <value = int64[0] {{}}> ()
+                i64_1 = Constant <value = int64[1] {{1}}> ()
+                i64_0 = Constant <value = int64[1] {{0}}> ()
+                i64_neg1 = Constant <value = int64[1] {{-1}}> ()
+
+                s_position = ArgMax<axis = -1, keepdims = 0>({self.input_names[1]})
+                e_position = ArgMax<axis = -1, keepdims = 0>({self.input_names[2]})
+                ee_position = Add(e_position,i64_1)
+                u_i64_neg1 = Unsqueeze(i64_neg1, i64_0)
+                slice_ids= Slice({self.input_names[0]}, s_position, ee_position, i64_neg1)
+                {self.output_names[0]} = {self._domain}.BertTokenizerDecoder (slice_ids, i64_em)
+            }}
+            """
+        )
+        bert_tokenizer_param = self._tokenizer_param
+
+        if isinstance(bert_tokenizer_param.vocab_or_file, (Path, str)):
+            # read from file
+            import json
+
+            with open(bert_tokenizer_param.vocab_or_file, "r") as f:
+                vocab = json.load(f)
+        else:
+            vocab = bert_tokenizer_param.vocab_or_file
+        ordered_vocab = OrderedDict(sorted(vocab.items(), key=lambda item: int(item[1])))
+        vocab = "\n".join(ordered_vocab.keys())
+        attrs = dict(vocab_file=vocab)
+        token_model_attr = []
+        for attr in attrs:
+            attr_value = onnx.helper.make_attribute(attr, attrs[attr])
+            token_model_attr.append(attr_value)
+
+        node_idx = next(i for i, v in enumerate(converter_graph.node) if v.op_type == "BertTokenizerDecoder")
+        converter_graph.node[node_idx].attribute.extend(token_model_attr)
+        return converter_graph
+
+
+class SequenceClassify(Step):
+    def __init__(self, name: Optional[str] = None):
+        """
+        Brief:
+            Convert max logit in logits array to index of classes, which used in classify task.
+        Args:
+            name: Optional name of step. Defaults to 'SequenceClassify'
+
+        """
+        super().__init__(["logits"], ["index"], name)
+
+    def _create_graph_for_step(self, graph: onnx.GraphProto, onnx_opset: int):
+        graph_input_names = [inp.name for inp in graph.output]
+
+        is_input_names_in_graph = [inp in graph_input_names for inp in self.input_names]
+        assert all(is_input_names_in_graph), f"Input names {self.input_names} not in graph inputs {graph_input_names}"
```

Comment:
Step.connect should have wired things up and self.input_names should always be present in graph.output so that individual Steps don't have to repeat that check (i.e. it should be done once in a central place instead of in every single Step._create_graph_for_step).